src/lightkurve/io/generic.py

```diff
@@ -37,7 +39,9 @@ def read_generic_lightcurve(
     # Raise an exception if the requested extension is invalid
     if isinstance(ext, str):
         validate_method(ext, supported_methods=[hdu.name.lower() for hdu in hdulist])
-    tab = Table.read(hdulist[ext], format="fits")
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=UnitsWarning)
```

Comment:
I propose documenting why we are ignoring warnings here as follows:

```suggestion
        # By default, AstroPy emits noisy warnings about units commonly used
        # in archived TESS data products (e.g., "e-/s" and "pixels").
        # We ignore them here because they don't affect Lightkurve's features.
        # Inconsistencies between TESS data products and the FITS standard
        # out to be addressed at the archive level. (See issue #1216.)
        warnings.simplefilter("ignore", category=UnitsWarning)
```